File: app/src/main/java/com/roman/gurdan/sudo/pro/data/dao/WeekDao.java
package com.roman.gurdan.sudo.pro.data.dao;

import androidx.annotation.NonNull;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.roman.gurdan.sudo.pro.data.entry.Game;
import com.roman.gurdan.sudo.pro.data.entry.Weekly;

import java.util.List;

@Dao
public interface WeekDao {

    @Query("select * from weekly")
    List<Weekly> getWeeks();

    @Query("select count(*) from weekly where date = :date")
    int getWeekly(String date);

    @Update
    void update(@NonNull Weekly item);

    @Delete
    void delete(@NonNull Weekly item);

    @Insert
    void insert(@NonNull Weekly item);

    @Query("select count(id) from weekly where result = :result")
    int getWeeksResult(int result);

    //近30天
    @Query("select count(id) from weekly where DATE('YYYY-MM-DD', '-15 day') <= :date and result = :result")
    int getWeekly(String date, int result);




}

File: app/src/main/java/com/roman/gurdan/sudo/pro/data/dao/WeekDao.kt
package com.roman.gurdan.sudo.pro.data.dao


import androidx.annotation.NonNull
import androidx.room.*
import com.roman.gurdan.sudo.pro.data.entry.Weekly


@Dao
interface WeekDao {

    @Query("select * from weekly")
    fun getWeeks(): List<Weekly>

    @Query("select count(*) from weekly where date = :date")
    fun getWeekly(date: String): Int

    @Update
    fun update(@NonNull item: Weekly)

    @Delete
    fun delete(@NonNull item: Weekly)

    @Insert
    fun insert(@NonNull item: Weekly)

    @Query("select count(id) from weekly where result = :result")
    fun getWeeksResult(result: Int): Int

    //近30天
    @Query("select count(id) from weekly where DATE('YYYY-MM-DD', '-15 day') <= :date and result = :result")
    fun getWeekly(date: String, result: Int): Int


}